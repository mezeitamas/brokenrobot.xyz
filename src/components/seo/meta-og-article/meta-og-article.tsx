import React from 'react';
import type { FunctionComponent } from 'react';

type MetaOgArticleProps = {
    siteName: string;
    title: string;
    description: string;
    url: string;
    published: string;
    author: string;
};

const MetaOgArticle: FunctionComponent<MetaOgArticleProps> = ({
    siteName,
    title,
    description,
    url,
    published,
    author
}) => {
    return (
        <>
            <meta
                property="og:type"
                content="article"
            />
            <meta
                property="og:locale"
                content="en_US"
            />
            <meta
                property="og:site_name"
                content={siteName}
            />
            <meta
                property="og:title"
                content={title}
            />
            <meta
                property="og:description"
                content={description}
            />
            <meta
                property="og:url"
                content={url}
            />
            <meta
                property="og:article:published_time"
                content={published}
            />
            <meta
                property="og:article:author"
                content={author}
            />
        </>
    );
};

export { MetaOgArticle };

export type { MetaOgArticleProps };
