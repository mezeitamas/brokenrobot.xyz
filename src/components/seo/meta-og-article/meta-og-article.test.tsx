import React from 'react';

import { create } from 'react-test-renderer';

import { MetaOgArticle } from './meta-og-article';
import type { MetaOgArticleProps } from './meta-og-article';

describe('<MetaOgArticle />', () => {
    test('should render', () => {
        // Given
        const props: MetaOgArticleProps = {
            siteName: 'Broken Robot',
            title: 'Hosting a static website on Amazon S3',
            description:
                'Hosting a simple static website on Amazon Web Services (AWS) can be daunting at first, but luckily itâ€™s quite straightforward. One option...',
            url: 'https://www.brokenrobot.xyz/blog/hosting-a-static-website-on-amazon-s3/',
            published: '2023-05-04T17:58:32.000Z',
            tags: ['aws', 's3', 'route53', 'staticwebsite', 'hosting'],
            author: 'Tamas Mezei'
        };

        // When
        const component = create(<MetaOgArticle {...props} />).toJSON();

        // Then
        expect(component).toMatchSnapshot();
    });
});
