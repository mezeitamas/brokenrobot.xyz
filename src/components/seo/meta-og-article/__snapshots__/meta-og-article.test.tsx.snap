// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MetaOgArticle /> should render 1`] = `
[
  <meta
    content="article"
    property="og:type"
  />,
  <meta
    content="en_US"
    property="og:locale"
  />,
  <meta
    content="Broken Robot"
    property="og:site_name"
  />,
  <meta
    content="Hosting a static website on Amazon S3"
    property="og:title"
  />,
  <meta
    content="Hosting a simple static website on Amazon Web Services (AWS) can be daunting at first, but luckily itâ€™s quite straightforward. One option..."
    property="og:description"
  />,
  <meta
    content="https://www.brokenrobot.xyz/blog/hosting-a-static-website-on-amazon-s3/"
    property="og:url"
  />,
  <meta
    content="2023-05-04"
    property="og:article:published_time"
  />,
  <meta
    content="Tamas Mezei"
    property="og:article:author"
  />,
]
`;
